### PR TITLE
Acquire version info with using platform details.

### PR DESCRIPTION
**Pull request by sersut**
In this PR we add the ability to acquire information from omnitruck without platfom specific info. We use the brand new `/versions` endpoint. 

@schisamo @patrick-wright @thommay

### CODE REVIEW

**Comment by sersut** [comment]
Argh... Forgot to add some tests for the list functionality. Stay tuned for that.

**Comment by sersut** [comment]
Allright tests are added. Sorry for the hiccup all..

File: lib/mixlib/install/backend/omnitruck.rb [hunk 1]
```diff
@@ -32,20 +33,33 @@ def initialize(options)
...
+            builds = omnitruck_get("versions", v: options.product_version)
+            ArtifactInfo.from_metadata_map(builds)
+          end
+        end
 
-          endpoint = "https://omnitruck.chef.io"
+        private
 
-          uri = URI.parse(endpoint)
+        def omnitruck_get(endpoint, parameters)
```
Comment (patrick-wright) [inline lib/mixlib/install/backend/omnitruck.rb]: endpoint should be resource

File: spec/mixlib/install/backend_spec.rb [hunk 2]
```diff
@@ -211,6 +293,44 @@
...
+        context "with a major product version" do
+          let(:product_version) { "12" }
+          let(:expected_version) { /^12.\d.\d\+[0-9]{14}$/ }
+
+          it_behaves_like "the right artifact list info"
+        end
+
+        context "with latest version keyword" do
+          let(:product_version) { "latest" }
+          let(:expected_version) { /\d.\d.\d/ }
```
Comment (patrick-wright) [inline spec/mixlib/install/backend_spec.rb]: This regexp should probably look like l:308 `/^12.4.3\+[0-9]{14}$/` (unrelated sorry, but I just noticed it)

**Comment by patrick-wright** [comment]
+1

**Comment by schisamo** [comment]
Overall looks good! I wonder if we should rename `ArtifactInfo` to `ArtifactMetadata`?

File: spec/mixlib/install/backend_spec.rb (hunk 2 above)
Comment (sersut) [inline spec/mixlib/install/backend_spec.rb]: Nice catch @patrick-wright

File: lib/mixlib/install/backend/omnitruck.rb (hunk 1 above)
Comment (sersut) [inline lib/mixlib/install/backend/omnitruck.rb]: :+1:

**Comment by sersut** [comment]
I like `ArtifactInfo` and `ArtifactMetadata` both :smil

**Comment by sersut** [comment]
Travis failures are `JSON.parse` failures that we get from omnitruck randomly even though we set the correct headers. I think we should merge this to unblock us for the `:unstable` channel work @patrick-wright

**Comment by sersut** [comment]
Merging this to unblock @patrick-wright on his changes. Will look into the JSON.parse issues right after.

**Pull request merged**